quota/etcd/storage/quota_storage.go

```diff
@@ -0,0 +1,387 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package storage contains storage classes for etcd-based quotas.
+package storage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/clientv3/concurrency"
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/proto"
+	"github.com/google/trillian/quota"
+	"github.com/google/trillian/quota/etcd/storagepb"
+)
+
+const (
+	configsKey = "quotas/configs"
+)
+
+var (
+	// timeNow is used in place of time.Now, so it may be replaced by tests.
+	timeNow = time.Now
+
+	globalPattern *regexp.Regexp
+	treesPattern  *regexp.Regexp
+	usersPattern  *regexp.Regexp
+)
+
+func init() {
+	var err error
+	globalPattern, err = regexp.Compile("quotas/global/(read|write)/config")
```

Comment:
Do you need the var? globalPattern, err := ... ? + below.

Input regex strings might be more visible as constants.